openapi_tester/schema_tester.py

```diff
@@ -264,7 +312,23 @@ def test_schema_section(
         """
         This method orchestrates the testing of a schema section
         """
-        if "oneOf" in schema_section and data is not None:
+        schema_section_type = schema_section.get("type")
+        if not schema_section_type and "properties" in schema_section:
+            schema_section_type = "object"
+        if schema_section_type is None or data is None and self.is_nullable(schema_section):
+            # If there`s no type, any response is permitted so we return early
+            # If data is None and nullable, we also return early
+            return
+        if data is None:
```

Comment:
not data

Reply:
Don't think this is right. An empty string, array, or object should have their schemas assessed - here we're checking nullability, so we're actually just after `None`s